src/Testing/LeanCode.IntegrationTestHelpers/LeanCodeTestFactory.cs

```diff
@@ -16,60 +15,57 @@ namespace LeanCode.IntegrationTestHelpers;
 public abstract class LeanCodeTestFactory<TStartup> : WebApplicationFactory<TStartup>, IAsyncLifetime
     where TStartup : class
 {
-    protected virtual ConfigurationOverrides Configuration { get; } = new ConfigurationOverrides();
-    protected virtual string ApiBaseAddress => "api/";
-    protected virtual string AuthBaseAddress => "auth/";
+    protected abstract ConfigurationOverrides Configuration { get; }
 
-    public string? CurrentUserToken { get; private set; }
+    public virtual JsonSerializerOptions JsonOptions { get; } =
+        new()
+        {
+            Converters =
+            {
+                new JsonLaxDateOnlyConverter(),
+                new JsonLaxTimeOnlyConverter(),
+                new JsonLaxDateTimeOffsetConverter(),
+            },
+        };
 
-    public virtual HttpClient CreateApiClient()
+    protected virtual string ApiBaseAddress => "api/";
+
+    public virtual HttpClient CreateApiClient(Action<HttpClient>? config = null)
     {
         var apiBase = UrlHelper.Concat("http://localhost/", ApiBaseAddress);
         var client = CreateDefaultClient(new Uri(apiBase));
-        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", CurrentUserToken);
+        config?.Invoke(client);
         return client;
     }
 
-    public virtual HttpClient CreateAuthClient()
+    public virtual HttpQueriesExecutor CreateQueriesExecutor(Action<HttpClient>? config = null)
     {
-        var apiBase = UrlHelper.Concat("http://localhost/", AuthBaseAddress);
-        return CreateDefaultClient(new Uri(apiBase));
+        return new HttpQueriesExecutor(CreateApiClient(config), JsonOptions);
     }
 
-    public virtual HttpQueriesExecutor CreateQueriesExecutor()
+    public virtual HttpCommandsExecutor CreateCommandsExecutor(Action<HttpClient>? config = null)
     {
-        return new HttpQueriesExecutor(CreateApiClient());
+        return new HttpCommandsExecutor(CreateApiClient(config), JsonOptions);
     }
 
-    public virtual HttpCommandsExecutor CreateCommandsExecutor()
+    public virtual HttpOperationsExecutor CreateOperationsExecutor(Action<HttpClient>? config = null)
     {
-        return new HttpCommandsExecutor(CreateApiClient());
+        return new HttpOperationsExecutor(CreateApiClient(config), JsonOptions);
     }
 
-    public virtual async Task<bool> AuthenticateAsync(PasswordTokenRequest tokenRequest)
+    public virtual async Task WaitForBusAsync(TimeSpan? delay = null)
     {
-        // FIXME: what if the auth server is outside of the app?
-        using var client = CreateAuthClient();
+        delay ??= TimeSpan.FromSeconds(5);
```

Comment:
Assign to another local of type `TimeSpan` to avoid awkward `.Value` later.

Reply:
Don't see anything awkward in it

Reply:
I do, you're kind of relying on the compiler to optimize away null checks.